src/main/java/org/broadinstitute/hellbender/tools/GatherVcfs.java

```diff
@@ -324,6 +368,10 @@ private static void gatherWithBlockCopying(final List<File> vcfs, final File out
                                 lastByteNewline = thisByteNewline;
                             }
 
+                            if( firstNonHeaderByteIndex == -1 ){
+                                log.warn("Scanned the entire file " + f.toUri().toString() + " and found no variants");
```

Comment:
It seems to me that at this point you have not necessarily scanned the entire file